Browser_Actions/Browser_ActionTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SeleniumExtras.WaitHelpers;
+using OpenQA.Selenium.Interactions;
+using System.Xml.Linq;
+using System.Drawing;
+using OpenQA.Selenium.Chrome;
+
+namespace Browser_Actions
+{
+    [TestFixture]
+    public class Browser_ActionTests : BaseTest_BA
+    {
+        [Test]
+        public void OpenNewWindowHandleTabs()
+        {
+            Setup();
+            string originalWindow = Driver.CurrentWindowHandle;
+            Assert.AreEqual(Driver.WindowHandles.Count, 1);
+            var elementMultipleWindows = Driver.FindElement(By.CssSelector("[href*='windows']"));
+            elementMultipleWindows.Click();
+            var elementClickHere = Driver.FindElement(By.XPath("//*[@class='example']//a[1]"));
+            elementClickHere.Click();
+            var wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(20));
+            wait.Until(wd => wd.WindowHandles.Count == 2);
+            foreach (string window in Driver.WindowHandles)
+            {
+                if (originalWindow != window)
+                {
+                    Driver.SwitchTo().Window(window);
+                    break;
+                }
+            }
+            wait.Until(wd => wd.Url == $"{MainUrl}/windows/new");
+            var elementContent = Driver.FindElement(By.ClassName("example"));
+            Assert.That(elementContent.Displayed);
+        }
+
+        [Test]
+        public void NavigateBackForward()
+        {
+            Setup();
+            var elementFormAuthentication = Driver.FindElement(By.CssSelector("[href*='login']"));
+            elementFormAuthentication.Click();
+            var elementUsername = Driver.FindElement(By.Id("username"));
+            elementUsername.SendKeys("tomsmith");
+            var elemenPassword = Driver.FindElement(By.Id("password"));
+            elemenPassword.SendKeys("SuperSecretPassword!");
+            var elementLoginButton = Driver.FindElement(By.XPath("//*[@class='radius']"));
+            elementLoginButton.Click();
+            var elementLogoutButton = Driver.FindElement(By.CssSelector("[href*='logout']"));
+            elementLogoutButton.Click();
+            Driver.Navigate().Back();
```

Comment:
Please add check between Driver.Navigate().Back(); and Driver.Navigate().Forward(); that needed page was opened

Reply:
did that